apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx

```diff
@@ -59,14 +60,44 @@ const QuizForm: React.FC<Properties> = ({ onNext }: Properties) => {
 		void dispatch(quizActions.previousQuestion());
 	}, [dispatch]);
 
-	const handleNextStep = useCallback(() => {
-		if (isLast) {
-			return;
-		}
+	const checkIfValueIsUndefined = useCallback(
+		(formData: FormToSave, keysToCheck: string[]): boolean => {
+			return keysToCheck.some((key) => formData[key] === undefined);
+		},
+		[],
+	);
+
+	const getAnswerIds = useCallback((formData: FormToSave) => {
+		return Object.values(formData).map(Number);
+	}, []);
+
+	const handleNextStep = useCallback(
+		(data: QuizFormValues) => {
+			if (category) {
+				const ids = category.map(
+					(categoryItem) => `answer${categoryItem.id.toString()}`,
+				);
+				const hasUndefined = checkIfValueIsUndefined(data, ids);
+
+				if (!hasUndefined) {
```

Comment:
I don't really understand this logic, could you explain?

Reply:
Previous and next question is actually handled on redux state. So initially the form with input names with the current question id is undefined so they cannot proceed to the next question. But when they go back and since the next question is already on redux state, I still need to check the form if it is undefined.

to make it a bit clearer I'll refactor some names to aligned to what it is doing.